src/main/scala/cognite/spark/v1/DefaultSource.scala

```diff
@@ -85,13 +94,114 @@ class DefaultSource
     new SequenceRowsRelation(config, sequenceId)(sqlContext)
   }
 
+  private def getFunctionResult(
+      client: GenericClient[IO],
+      functionId: Long,
+      result: FunctionCall): IO[Json] = {
+    var res = result
+    while (res.status != None && res.status.get == "Running") {
+      res = client
+        .functionCalls(functionId)
+        .retrieveById(res.id.get)
+        .unsafeRunSync()
+    }
+
+    if (res.status != None && res.status.get == "Completed") {
+      for {
+        functionCallResponse <- client
+          .functionCalls(functionId)
+          .retrieveResponse(res.id.getOrElse(0L))
+      } yield functionCallResponse.response.getOrElse(Json.fromJsonObject(JsonObject.empty))
+    } else {
+      IO.pure(Json.fromJsonObject(JsonObject.empty))
```

Comment:
Returning an empty Json object on errors seems a bit strange, I think it might be better to throw an exception.